src/main/java/org/jabref/gui/editmode/EditModeAction.java

```diff
@@ -0,0 +1,47 @@
+package org.jabref.gui.editmode;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.database.BibDatabaseMode;
+
+import java.util.Optional;
+
+import static org.jabref.gui.actions.ActionHelper.needsDatabase;
+
+public class EditModeAction extends SimpleCommand {
```

Comment:
I would suggest renaming it to EditBibDatabaseMode to have it a bit more clear which mode is meant.
Otherwise lgtm! Thanks  for your ongoing contribution!